tests/single/gas/test_pre_verification_gas_calculation.py

```diff
@@ -0,0 +1,110 @@
+from dataclasses import asdict
+import pytest
+
+from tests.types import UserOperation, RPCRequest
+from tests.utils import assert_ok, deploy_contract
+
+
+# create a copy of a UserOperation object while overriding a 'test_field_name' field with a 'new_value'
+def set_user_op_field_value(user_op_input, test_field_name, new_value):
+    user_op = asdict(user_op_input)
+    user_op[test_field_name] = hex(new_value)
+    return UserOperation(**user_op)
+
+
+# perform a binary search for a minimal valid numeric value for a UserOperation field
+def find_min_value_for_field(
+    entrypoint_contract, user_op_input, test_field_name, minimum_value, maximum_value
+):
+    # check that maximum value is sufficient
+    assert_ok(RPCRequest(method="debug_bundler_clearState").send())
+    user_op = set_user_op_field_value(user_op_input, test_field_name, maximum_value)
+    res = user_op.send()
+    if hasattr(res, "message") and res.message is not None:
+        raise ValueError(
+            f"providing '{test_field_name}' with maximum_value={maximum_value} should succeed but reverted with: {res.message}"
+        )
+
+    # check that minimum value is insufficient
+    assert_ok(RPCRequest(method="debug_bundler_clearState").send())
+    user_op = set_user_op_field_value(user_op, test_field_name, minimum_value)
+    res = user_op.send()
+    if not hasattr(res, "message") or res.message is None:
+        raise ValueError(
+            f"providing '{test_field_name}' with minimum_value={minimum_value} should revert but it succeeded"
+        )
+
+    low = minimum_value
+    high = maximum_value
+    while low < high:
+        mid = low + (high - low) // 2
+        print(f"low:{low} / high: {high} / mid: {mid}")
+        assert_ok(RPCRequest(method="debug_bundler_clearState").send())
+        # operation may have been included - update nonce for the next check
+        user_op = set_user_op_field_value(user_op_input, test_field_name, mid)
```

Comment:
how it can be included? we're running the bundler in "manual mode"

Reply:
This test does not run in manual mode and may include these userops.